### ISSUE TITLE
Build fails on macOS after e5c8bee85e7ae2d974069d63528333ffacc1923b

### ISSUE THREAD

**Issue by rex4539** [ocaml/ocaml#9528]
After e5c8bee85e7ae2d974069d63528333ffacc1923b was merged, build fails on macOS.

```
clang: error: no such file or directory: 'misc.b.c'
clang: error: no input files
clang: error: no such file or directory: 'interp.b.c'
clang: error: no input files
make[2]: *** [misc.b.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [interp.b.o] Error 1
clang: error: no such file or directory: 'stacks.b.c'
clang: error: no input files
make[2]: *** [stacks.b.o] Error 1
clang: error: no such file or directory: 'fix_code.b.c'
clang: error: no input files
make[2]: *** [fix_code.b.o] Error 1
clang: error: no such file or directory: 'startup_aux.b.c'
clang: error: no input files
make[2]: *** [startup_aux.b.o] Error 1
clang: error: no such file or directory: 'startup_byt.b.c'
clang: error: clangno input files: 
error: no such file or directory: 'major_gc.b.c'
clang: clang: error: error: no such file or directory: 'freelist.b.c'no input files

clang: error: no input files
make[2]: *** [major_gc.b.o] Error 1
make[2]: *** [freelist.b.o] Error 1
make[2]: *** [startup_byt.b.o] Error 1
clang: error: no such file or directory: 'minor_gc.b.c'
clang: error: no input files
clang: error: no such file or directory: 'memory.b.c'
clang: error: no input files
make[2]: *** [minor_gc.b.o] Error 1
make[2]: *** [memory.b.o] Error 1
clang: error: no such file or directory: 'alloc.b.c'
clang: error: no input files
clang: error: no such file or directory: 'roots_byt.b.c'
clang: error: no input files
make[2]: *** [alloc.b.o] Error 1
make[2]: *** [roots_byt.b.o] Error 1
make[1]: *** [coldstart] Error 2
make: *** [world.opt] Error 2
```

**Comment by dra27** [comment]
Which version of `make` are you using?

**Comment by rex4539** [comment]
```
% make --version
GNU Make 3.81
```

**Comment by dra27** [comment]
OK, so the 14 year old version of make interprets pattern rules differently! There are two immediate workarounds:

- Use homebrew or macports to get GNU make 4
- Configure with `--disable-dependency-generation` (except presumably you're actually working on the compiler?)

**Comment by rex4539** [comment]
Thanks!

**Comment by dra27** [comment]
I'm glad that fixes it, but I'll keep the issue open for now - I'm just investigating whether there's a permanent fix!

**Comment by dra27** [comment]
I fired up my MacBook and reproduced the problem - permanent fix in #9529
